rust/src/builder.rs

```diff
@@ -153,6 +170,17 @@ impl DeltaTableBuilder {
         self
     }
 
+    /// Sets `log_buffer_size` to the builder
+    pub fn with_buffer(mut self, log_buffer_size: usize) -> DeltaResult<Self> {
+        if log_buffer_size > 50 {
+            return Err(DeltaTableError::Generic(String::from("Log buffer size should smaller than 51")));
```

Comment:
Why make this limit? Practically I don't think many will want to increase this, but seems fine to not have an explicit limit. In most cases, because of checkpoints there is already an implicit limit on the number of calls.

Reply:
Yep, this was relevant in the earlier version where max_version was not known, because in that case you would have 49 wasted requests if there was only one file. But I removed that now.